desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,264 @@
+"""Objectives for targeting neoclassical transport."""
+
+import numpy as np
+
+from desc.compute import get_profiles, get_transforms
+from desc.compute.utils import _compute as compute_fun
+from desc.grid import LinearGrid
+from desc.utils import Timer
+
+from ..integrals.quad_utils import chebgauss2
+from .objective_funs import _Objective
+from .utils import _parse_callable_target_bounds
+
+
+class EffectiveRipple(_Objective):
+    """The effective ripple is a proxy for neoclassical transport.
+
+    The 3D geometry of the magnetic field in stellarators produces local magnetic
+    wells that lead to bad confinement properties with enhanced radial drift,
+    especially for trapped particles. Neoclassical (thermal) transport can become the
+    dominant transport channel in stellarators which are not optimized to reduce it.
+    The effective ripple is a proxy, measuring the effective modulation amplitude of the
+    magnetic field averaged along a magnetic surface, which can be used to optimize for
+    stellarators with improved confinement. It is targeted as a flux surface function.
+
+    References
+    ----------
+    https://doi.org/10.1063/1.873749.
+    Evaluation of 1/ν neoclassical transport in stellarators.
+    V. V. Nemov, S. V. Kasilov, W. Kernbichler, M. F. Heyn.
+    Phys. Plasmas 1 December 1999; 6 (12): 4622–4632.
+
+    Parameters
```

Comment:
this stuff needs to be updated with the new doc api

Reply:
Done on #1290